File: internal/view/component/logs.go
package component

import (
	"encoding/json"
)

type LogsConfig struct {
	Namespace  string   `json:"namespace,omitempty"`
	Name       string   `json:"name,omitempty"`
	Containers []string `json:"containers,omitempty"`
}

type Logs struct {
	Metadata Metadata   `json:"metadata,omitempty"`
	Config   LogsConfig `json:"config,omitempty"`
}

func NewLogs(namespace, name string, containers []string) *Logs {
	return &Logs{
		Config: LogsConfig{
			Namespace:  namespace,
			Name:       name,
			Containers: containers,
		},
		Metadata: Metadata{
			Type: "logs",
			Title: []TitleViewComponent{
				NewText("Logs"),
			},
		},
	}
}

// GetMetadata accesses the components metadata. Implements ViewComponent.
func (l *Logs) GetMetadata() Metadata {
	return l.Metadata
}

type logsMarshal Logs

func (l *Logs) MarshalJSON() ([]byte, error) {
	m := logsMarshal(*l)
	m.Metadata.Type = "logs"

	return json.Marshal(&m)
}

File: web/react/src/models/logs.ts
import { LogsModel, TitleView, toTitle } from 'models'

export class JSONLogs implements LogsModel {
  readonly type = 'logs'
  readonly title: TitleView
  readonly namespace: string
  readonly name: string
  readonly containers: string[]
  readonly accessor: string

  constructor(ct: ContentType) {
    if (ct.metadata.title) {
      this.title = toTitle(ct.metadata.title)
    }

    this.accessor = ct.metadata.accessor
    this.namespace = ct.config.namespace
    this.name = ct.config.name
    this.containers = ct.config.containers
  }
}
